elaborate on edge types in relay note (#378)
apollographql/docs: src/content/technotes/TN0029-relay-style-connections.mdx

Consider the ["Zero, One, Infinity" rule](https://en.wikipedia.org/wiki/Zero_one_infinity_rule) — can you definitively assert that your list will never require pagination or other metadata about the relationship?

> Using a wrapper type for lists provides two benefits:

Reviewer: ```suggestion
Using a wrapper type for lists provides the following benefits:
```

Reviewer: Old documentarian's trick: never say how many items are in the list, in case you later add another item to the list 😄

> 1. **Easier evolution**: you can return a wrapper type with no pagination in place today and then add pagination without breaking existing clients. If you return a list directly, you can't add pagination metadata later.

Reviewer: ```suggestion
- **Avoid breaking changes.** You can initially return a wrapper type that _doesn't_ use pagination, and then add pagination later without breaking existing clients. If you return a list directly, you _can't_ add pagination metadata later.
```

Reviewer: (We only use ordered lists for sequences of steps)

> 2. **Modeling relationships**: The `Connection` and `Edge` wrapper types support fields that model attributes _of the relationship between entities_ that don't belong in the entities themselves. Consider this example of a many-to-many relationship between `Business` and `Customer`:

Reviewer: ```suggestion
- **Represent entity relationships.** The `Connection` and `Edge` wrapper types support fields that model attributes _of the relationship between entities_ that don't belong in the entities themselves.

    Consider this example of a many-to-many relationship between `Business` and `Customer`:
```

>    ```graphql {13}

Reviewer: ```suggestion
   ```graphql
```

Reviewer: Also I think this line and every added line below needs _one_ more leading space to make it 4 spaces so it all properly indents within the list item?

Author: It indents properly — prettier picked the indentation level. it looks correct in the rendered version

   type Business {
     id: ID
     customers: CustomerConnection
   }

   type CustomerConnection {
     edges: [CustomerEdge]
     total: Int
   }

   type CustomerEdge {
     node: Customer

>      type: CustomerType

Reviewer: ```suggestion
     type: CustomerType #highlight-line
```

Reviewer: We're generally getting away from line-number-based highlighting to reduce later breaks

   }

   enum CustomerType {
     IN_STORE
     ONLINE
     MULTI_CHANNEL
   }

   type Customer {
     id: ID
     shopsAt: BusinessConnection # --snip --
   }
   ```

>    A specific `Customer` may shop at one business `IN_STORE` and another `ONLINE`. The `type` is a attribute of the _relationship_, not the business or customer themselves. Without wrapper types, you don't have a place to put this data.

Reviewer: ```suggestion
   A specific `Customer` might shop at one business `IN_STORE` and another `ONLINE`. The `type` is an attribute of the _relationship_, not the business or customer itself. Without wrapper types, you don't have a place to put this data.
```

Author: i need something in my editor that highlights every time i use "may"
